### PR TITLE
Update 22-query.sh

### PR DESCRIPTION
**Pull request by gavindi**
Created function that counts tool use by each user belonging to a specified group and within the specified month

### CODE REVIEW

File: parts/22-query.sh [hunk 1]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
 	EOF
 }
 
+local_query-toolusebygroup() { ## : <year-month> <group> Lists count of tools used by all users in a group
```
Comment (hexylena) [inline parts/22-query.sh:955]:
```suggestion
query_tool-use-by-group() { ## : <year-month> <group> Lists count of tools used by all users in a group
```

local is only used for local commands, query must be followed by a `_`, and then generally we use `-` to separate words in a tool command

File: parts/22-query.sh [hunk 2]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+        handle_help "$@" <<-EOFhelp
+                Lists tools use count by users in group.
+                Requires <year-month> (2022-03) and <group> 
+
+                Example:
+                $ gxadmin local query-toolusebygroup 2022-02 CSIRO
+                tool_id                                             |             username             | count 
+                ----------------------------------------------------+----------------------------------+-------
+                CONVERTER_gz_to_uncompressed                        | erin_hahn                        |     1
```
Comment (hexylena) [inline parts/22-query.sh:964]: Could you replace these names with 'fake' names? Joe Bloggs / jane doe / etc. Generally we try not to include PII in documentation

File: parts/22-query.sh [hunk 3]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
 	EOF
 }
 
+local_query-toolusebygroup() { ## : <year-month> <group> Lists count of tools used by all users in a group
+        handle_help "$@" <<-EOFhelp
```
Comment (hexylena) [inline parts/22-query.sh:956]: please use tabs for indentation (this is why the test is failing)

File: parts/22-query.sh (hunk 1 above)
Comment (hexylena) [reply parts/22-query.sh:955]:
```suggestion
local_query-toolusebygroup() { ##? <year_month> <group>: Lists count of tools used by all users in a group
```
The : needs to come at the end. Additionally I'm making a suggestion to use `##?` which invokes our automatic command line parsing, saving you the bother of doing error checking.

File: parts/22-query.sh [hunk 4]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+                Cut1                                                | Benson, Derek (IM&T, Pullenvale) |     1
+                Cut1                                                | jodie.vandekamp_at_csiro.au      |     1
+
+
+EOFhelp
+
+if (( $# > 0 )); then
+                wheredate=" date_trunc('month', job.create_time) = '$1-01'"
+                wheregroup="'$2'"
+fi
```
Comment (hexylena) [inline parts/22-query.sh:975]:
```suggestion
```

by using `##?` above, you're given automatic variables to use `$arg_year_month` and `$arg_group`, so we can remove this part compltely.

File: parts/22-query.sh [hunk 5]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+                WHERE
+                        job.user_id = galaxy_user.id
+                AND
+                        user_group_association.group_id = galaxy_group.id
+                AND
+                        user_group_association.user_id = galaxy_user.id
+                AND
+                        galaxy_group.name = $wheregroup
+                AND
+                        $wheredate
```
Comment (hexylena) [inline parts/22-query.sh:990]:
```suggestion
                        galaxy_group.name = '$arg_group'
                AND
                        date_trunc('month', job.create_time) = '$arg_year_month'
```

File: parts/22-query.sh [hunk 6]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+                Cut1                                                | jodie.vandekamp_at_csiro.au      |     1
+
+
+EOFhelp
+
+if (( $# > 0 )); then
+                wheredate=" date_trunc('month', job.create_time) = '$1-01'"
+                wheregroup="'$2'"
+fi
+        read -r -d '' QUERY <<-EOF
```
Comment (hexylena) [inline parts/22-query.sh:976]:
```suggestion
        username=$(gdpr_safe galaxy_user.username username "Anonymous User")
        read -r -d '' QUERY <<-EOF
```

For queries that obtain user information, we have a function to help us keep to GDPR compliance and hide unnecessary usernames. While not relevant for your case (judging by the function documentation) it is for the rest of our users especially if they want to share results with e.g. colleagues.

File: parts/22-query.sh [hunk 7]
```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+                SELECT
+                        job.tool_id, galaxy_user.username, count(job.tool_id)
```
Comment (hexylena) [inline parts/22-query.sh:978]:
```suggestion
                        job.tool_id, $username, count(job.tool_id)
```

in connection with the above comment

**Review by hexylena — COMMENTED** [review]
Hey @gavindi thanks for contributing this! I've made a number of comments to help it fit a bit better with gxadmin's features and style with explanations, please apply those changes at your convenienc.

It's great to see more contributions! This looks like a useful query!

Could you please also add it to CHANGELOG.md?

File: parts/22-query.sh (hunk 1 above)
Comment (gavindi) [reply parts/22-query.sh]: Hi hexylena.  As per your suggestion, this is now fixed.

File: parts/22-query.sh (hunk 4 above)
Comment (gavindi) [reply parts/22-query.sh]: Resolved as suggested.

File: parts/22-query.sh (hunk 6 above)
Comment (gavindi) [reply parts/22-query.sh]: Resolved as suggested.

File: parts/22-query.sh (hunk 7 above)
Comment (gavindi) [reply parts/22-query.sh]: Resolved as suggested.

**Comment by gavindi** [comment]
All done and thanks for your suggestions.  Can you please review?

File: parts/22-query.sh [hunk 8]
```diff
@@ -952,6 +952,43 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
...
+                Example:
+                $ gxadmin query tool-use-by-group 2022-02 NameOfGroup
+                tool_id                                             |             username             | count 
+                ----------------------------------------------------+----------------------------------+-------
+                CONVERTER_gz_to_uncompressed                        | user_1                           |     1
+                Convert characters1                                 | user_2                           |     1
+                Cut1                                                | user_2                           |     1
+                Cut1                                                | user_3                           |     1
+
+
```
Comment (hexylena) [inline parts/22-query.sh:969]: Almost done! Just replace the leading spaces here with tabs (2 should be good) and the tests will pass.

Then we can merge! Thanks again for this contribution. 

(I would make this change myself but you've made your PR from the `main` branch which I as a maintainer cannot contribute to. In the future if you make a branch before starting your PR then I can make some of the stylistic fixes myself and save ya some time.)

Comment (gavindi) [reply parts/22-query.sh]: Thank for bearing with me on this.  I'll get more proficient with this as I go.  I've now replaces the leading spaces with tabs.

Comment (hexylena) [reply parts/22-query.sh]: yea no dramas!

**Comment by hexylena** [comment]
Thanks for bearing with the changes @gavindi !

**Pull request merged**